### ISSUE TITLE
reCAPTCHA not showing

### ISSUE THREAD

**Issue by brashrebel** [easydigitaldownloads/edd-checkout-fields#129]
Reported by two customers ( [1](https://secure.helpscout.net/conversation/365209220/0/) and [2](https://secure.helpscout.net/conversation/357813637/58404/?folderId=180504) ) and replicated on my local site. The reCAPTCHA field does not appear on checkout, only the label. It also appears that the api.js script from Google is not actually getting enqueued.

**Comment by pippinsplugins** [comment]
@brashrebel My customer didn't have a problem with the field not showing but rather a problem with the shown reCaptcha never validating.

**Comment by chriscct7** [comment]
Out of date recaptcha API keys would be my guess. If they had them for a while, the old ones would have started to expire. You had to get new keys when recaptcha 2.0 came along.

**Comment by pippinsplugins** [comment]
My customer re-generated their keys. No bueno.

**Comment by chriscct7** [comment]
What's the url?

**Comment by brashrebel** [comment]
I had exactly the same experience as the customer in 1 above. I made brand new keys and setup on my local but the api.js script was not loaded and no reCAPTCHA appeared.

**Comment by chriscct7** [comment]
What happens if you try to browse to https://www.google.com/recaptcha/api/fallback?k={yourapikey}

**Comment by brashrebel** [comment]
> Sorry, something went wrong. Please try reloading the page.

**Comment by pippinsplugins** [comment]
Problem only happens when multiple gateways are enabled. Pretty sure it's due to the reCaptcha JS getting enqueued inline.

**Comment by pippinsplugins** [comment]
I have it rendering now but I cannot get it to validate because Google always returns a `timeout-or-duplicate` error. Haven't tracked down the cause yet.

**Comment by pippinsplugins** [comment]
@brashrebel could you give that a try?

**Comment by brashrebel** [comment]
Now the opposite occurs. When multiple gateways are present it works. Only one results on no visible reCAPTCHA. In my test I'm trying a paid product which does bring up two gateway options and then a free product which only brings up one. The latter shows nothing but the label. On master it is the opposite.

**Comment by pippinsplugins** [comment]
Bah!

**Comment by cklosowski** [comment]
We ran into this with EDD's use of `jQuery.payment` for the card formatting. The problem is that EDD does not trigger the `edd_gateway_loaded` javascript event if there is only a single gateway, or the gateway is defaulted. It will work if you have two gateways, and change to the other gateway because that is what triggers the event.

@brashrebel @pippinsplugins try this again on EDD 2.8.2 or later:
https://github.com/easydigitaldownloads/easy-digital-downloads/blob/master/assets/js/edd-ajax.js#L378

**Comment by cklosowski** [comment]
@brashrebel @pippinsplugins give this another try.

**Comment by brashrebel** [comment]
Pulled this branch again and now checking out results in:
`Notice: Undefined property: stdClass::$sucess in .../edd.dev/wp-content/plugins/edd-checkout-fields/classes/fields/recaptcha.php on line 156`

**Comment by brashrebel** [comment]
Oh, it's simply a misspelled property. I just fixed it and pushed the change to `issue/129`. Now it works perfectly. Nice work Chris!
